lib/spack/spack/solver/asp.py

```diff
@@ -1880,8 +1880,9 @@ def define_version_constraints(self):
         """Define what version_satisfies(...) means in ASP logic."""
         for pkg_name, versions in sorted(self.version_constraints):
             # version must be *one* of the ones the spec allows.
+            # Also, "possible versions" contain only concrete versions, so satisfies is appropriate
             allowed_versions = [
-                v for v in sorted(self.possible_versions[pkg_name]) if v.intersects(versions)
+                v for v in sorted(self.possible_versions[pkg_name]) if v.satisfies(versions)
```

Comment:
To be clear, in this point where `v.concrete is True` using `satisfies` or `intersects` is conceptually the same. `GitVersion.intersects` has a bug that require some time to be fixed, changing the version being called seems not ot incur in #36134